### ISSUE TITLE
derby-auth implementation - passport memory leak?

### ISSUE THREAD

**Issue by lefnire** [jaredhanson/passport#79]
Hey Jared,

I recently implemented auth for [Derby](http://derbyjs.com/) using Passport. It was originally implemented as Everyauth, but due to technicalities (later) had to be migrated to Passport. In the end, however, my server running the same logic as Everyauth, migrated to Passport, [started crashing](https://github.com/lefnire/derby-auth/issues/2) with "memory limit exceeded" errors (500mb).

I'm wondering if:
1. There's anything inherent of Passport that's less conducive to realtime applications than traditional apps (eg, session management over socket.io?)
2. If not, there's anything you can spot at a glance in the [derby-auth](https://github.com/lefnire/derby-auth/blob/master/index.js) codebase that's breaking Passport rules which may be causing memory leaks or other such performance issues.

TL;DR Backstory:
At first I chose Everyauth because Derby was created by the author of Everyauth - so it's a natural fit; and the Derby folks are using Everyauth themselves internally. However, I found that (1) Everyauth password authentication specifically ("Local Strategy") was less suited for Derby purposes than Passport, and (2) Passport lets us pass in strategy objects for each auth strategy we want to implement, rather than having to implement all the strategies manually in code.

Once that was built out, I replaced all the auth code for [my application](https://github.com/lefnire/habitrpg) using derby-auth and `git push heroku master`'d. The result was the app started crashing due to "memory limit exceeded" errors. The app worked fine when not logged in, but as soon as I logged in - refreshing once or twice crashed the app. Here's the original [Everyauth implementation](https://github.com/lefnire/derby-auth/tree/everyauth) if that's any useful reference.  I don't mean this to be a "fix my code" support request, but was hoping you have any top-of-the-dome knowledge so I don't go too far down a rabbit hole if it's not the right direction.
